apps/chatbot/src/bot/commands/notification.ts

```diff
@@ -0,0 +1,302 @@
+
+    if (param === "--push") {
+      isPush = true;
+      return false;
+    }
+
+    if (param === "--no-discord") {
+      isDiscord = false;
+      return false;
+    }
+
+    if (param === "--no-push") {
+      isPush = false;
+      return false;
+    }
+
+    return true;
+  });
+
+  return { isDiscord, isPush, imageUrl, restParams };
+}
+
+function parseOptionalLinkUrl(broadcasterName: string, params: string[]) {
+  let linkUrl = `https://twitch.tv/${broadcasterName}`;
+  let restParams = params;
+  const maybeUrl = safeParsePartialURL(params[0]);
+  if (maybeUrl && !partialUrlFilterList.includes(maybeUrl)) {
+    linkUrl = maybeUrl;
+    restParams = params.slice(1);
+  }
+  return { linkUrl, restParams };
+}
+
+function parseOptionalStartAndEnd(params: string[]): {
+  start?: DateStr;
+  end?: DateStr;
+  restParams: string[];
+} {
+  // YYYY-MM-DD[ HH:II] YYYY-MM-DD[ HH:II]
+  if (params[0] && params[0].match(/^\d{4}-\d{2}-\d{2}/)) {
+    let endDateString;
+    let startDateString = params.shift();
+    if (params[0] && params[0].match(/^\d{2}:\d{2}/)) {
+      startDateString += "T" + params.shift();
+    }
+
+    if (params[0] && params[0].match(/^\d{4}-\d{2}-\d{2}/)) {
+      endDateString = params.shift();
+
+      if (params[0] && params[0].match(/^\d{2}:\d{2}/)) {
+        endDateString += "T" + params.shift();
+      }
+    }
+
+    return {
+      start: startDateString as DateStr,
+      end: endDateString as DateStr,
+      restParams: params,
+    };
+  }
+
+  return { restParams: params };
+}
+
+function parseTitleAndText(params: string[]) {
+  const textParts = params.join(" ").split(/[|\/]/);
+  const title = textParts[0]?.trim();
+  const text = textParts.slice(1).join("/").trim();
+  return { title, text };
+}
+
+function renderPendingNotification(notification: PendingNotification) {
+  return [
+    notification.tag,
+    notification.start,
+    notification.end,
+    notification.title,
+    notification.text,
+    notification.linkUrl,
+    notification.isPush ? "PUSH" : null,
+    notification.isDiscord ? "DISCORD" : null,
+  ]
+    .filter((a) => a?.trim())
+    .join(" | ");
+}
+
+export async function createNotificationCommands() {
+  const pendingNotifications = new Map<
+    PendingNotificationKey,
+    PendingNotification
+  >();
+
+  const notificationCommand: CommandHandler = async (
+    params,
+    { broadcasterId, broadcasterName, userName, userId, reply }
+  ) => {
+    const isMod = await checkUserIsAllowedToSendNotifications(userName);
+    if (!isMod) {
+      reply("mayaHalt you are not allowed to send notifications");
+      return;
+    }
+
+    const { restParams: paramsWithoutChannelOptions, ...options } =
+      parseOptions(params);
+    const { linkUrl, restParams } = parseOptionalLinkUrl(
+      broadcasterName,
+      paramsWithoutChannelOptions
+    );
+    let { title, text } = parseTitleAndText(restParams);
+
+    if (!title) {
+      reply("Waiting Creating notification...");
```

Comment:
seems unnecessary to send this. it can however take a bit to fetch the channel info